### DISCUSSION TITLE
[Feature Request]: Support for USB<->SPI Interfaces (FTDI, WCH) to connect LoRa chips

### DISCUSSION THREAD

**Discussion by h0lad** [meshtastic/firmware#5467] [Ideas]
### Platform

Linux Native

### Description

As the project now has Linux support the easiest way to attach a LoRa module to a standard x64 system would be to use one of these FTDI or WCH USB<->SPI adapter PCBs. These can be bought for cheap from Aliexpress ([FT2232HL](https://de.aliexpress.com/item/1005005373869189.html), [CH347](https://de.aliexpress.com/item/1005008041621349.html) etc.) for a few $ or €.

So the idea:
- Add support for the most popular FTDI chips like the FT2232HL or FT232H or WCH stuff like CH347.
- WCH chips are great and super cheap too. But they have closed source creep. [Some stuff got reverse engineered](https://github.com/981213/ch347-nor-prog).

This also maybe will make automatic testing way more easy as the entire platform can be executed and debugged on a native system.

**Comment by jp-bennett** [comment]
There is a user-space driver for the CH341a in development. See the commits in https://github.com/meshtastic/firmware/tree/ch341

Developers have hardware getting assembled right now to finish this support.

Once that support is finished, we'll have a decent platform if anybody else wants to write a userspace driver for one of the other USB chips.
